osu.Game.Tests/Skins/TestSceneLegacySkin.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Threading.Tasks;
+using System.IO;
+using System.Collections.Generic;
+using NUnit.Framework;
+using osu.Game.Skinning;
+using osu.Game.Audio;
+using osu.Game.Tests.Resources;
+using osu.Game.Tests.Visual;
+using osu.Framework.IO.Stores;
+using osu.Framework.Testing;
+using osu.Framework.Audio;
+using osu.Framework.Audio.Sample;
+
+namespace osu.Game.Tests.Skins
+{
+    [HeadlessTest]
+    public class TestSceneLegacySkin : OsuTestScene
```

Comment:
This testing is inadequate because it doesn't test the result of gameplay loading. You want to add a test to `TestSceneHitObjectSamples`.

I'm fairly certain the implementation in this PR (inside `LegacySkin`) will fail, since `SkinnableSound` will fall back to the default lazer sample store if `null` is returned.